### DISCUSSION TITLE
Catalog API request or equivalent to get all items/ASINs (not listings) for a Brand

### DISCUSSION THREAD

**Discussion by kevnhowe** [amzn/selling-partner-api-models#3829] [General]
I am trying to query all of our products in the Amazon Catalog for our Brand. The Catalog API Item Search method requires either an identifier or keyword search parameter in order to return values. Any keyword I choose, even the broadest, when paired with my Brand parameter, only returns a fraction of our total Items listed. I am trying to pull from the Catalog API, as opposed to the Listings API or Listing Reports, because it is the master record for ASINs. Listings are the master record for Listings, which is an instance of an Item.
Please let me know if there is anything I can do with the Catalog API, or other utilities to pull all Items for our Brand.

**Comment by Sourav-techlur** [comment]
Hello @kevnhowe did you find any solution i am also trying to get the products which belongs to my seller central account which has currently 1 product but the catalog/2022-04-01/items returns 89496 products. I think it returns all products from all the seller store not from the seller id provided in the api.

**Reply by RicardoAReyes** [reply]
hi all, I'm also very curious to know the answer to that question posted on this thread.

**Comment by chapmanjw** [comment]
This is not a feature that we support today. I will forward this to our business team for review and potential addition to our future roadmap. If you're able, providing business impact data may aid the business team on prioritization.

**Reply by DanielLanger** [reply]
@chapmanjw any update on this? our use case is being able to programmatically pull all GTINs for a given seller. As a third party tool that integrates with seller accounts, we don't know the GTINs (aka UPCs) ahead of time, and would love for this endpoint to simply allow us to query all catalog items for a seller so we can gather a list of UPCs.

**Reply by Mkallday** [reply]
@chapmanjw Please confirm my understanding; there is no endpoint for a Seller to retrieve their current listings?

**Reply by kevnhowe** [reply]
Business Impact is this - We have no definitive record of truth for our Brand ASINs from Amazon. We require an internal dataset of ASIN Master Data so that we can synchronize and validate our desired Catalog attributes and published properties. This also allows us to ensure there are not Brand ASINs on Amazon that are listed by other sellers. Listings are not a master record of the Catalog, as evidenced by the ASIN not being a primary key on Listing records.

**Reply by kevnhowe** [reply]
> @chapmanjw Please confirm my understanding; there is no endpoint for a Seller to retrieve their current listings?

@Mkallday this is not what I was asking for. There is an endpoint for Listings. I am asking for a parameter that allows me to return Catalog Items/ASINs for our Brand through the Catalog API

**Reply by weilinggu** [reply]
@kevnhowe Just to clarify, you are looking for listings items that are not only listed by you but also listed by other sellers that has a specific brand?

**Comment by linde9821** [comment]
We would also like to get all the asins (or skus) in order to generate our own reports. So this feature would be very much appreciated.
